binary-search-tree/ruby/inventory_management.rb

```diff
@@ -34,17 +34,36 @@ def initialize
 
   def search(id)
     # Comment out the line below and make changes here
-    raise "you must implement search"
+    # raise "you must implement search"
+    node = @root
+
+    if node == nil
+      return nil
+    elsif node.val.id == id
+      return node.val
+    elsif id < node.val.id
+      node = node.left
+    else
+      node = node.right
+    end
```

Comment:
The issue here is that no recursion is taking place. To better familiarise yourself with recursion please have a look at this introductory post https://obtainable-fontina-7ef.notion.site/What-even-is-recursion-d48bac405c5746f1a15dfdea92c6580e. It might help completing the task